src/power_grid_model/utils.py

```diff
@@ -301,3 +304,56 @@ def _compatibility_deprecated_import_json_data(json_file: Path, data_type: Datas
         raise PowerGridSerializationError("An internal error occured during deserialization")
 
     return result
+
+
+def self_test():
+    """
+    Perform a self-test of the Power Grid Model functionality to check if the installation was successful
+    and there are no build errors, segfaults, undefined symbols, etc.
+
+    This function is designed to validate the basic functionality of data serialization,
+    model instantiation, power flow calculation, and serialization of results using the
+    Power Grid Model library.
+
+    Raises:
+        AssertionError: If the output data is empty or if the calculation results are incorrect.
```

Comment:
Maybe also add `PowerGridError` and/or `PowerGridSerializationError` (if i remember correctly) cause that's what's being raised if the other commands fail. Alternatively, you need to put a `try`/`except` block around it and raise it as an `AssertionError` but then you will have to make sure the info on what went wrong is not lost somehow. I think the former is easier

Reply:
the try/except block catches this
```suggestion
```